src/Components/Components/src/Forms/InputComponents/InputDate.cs

```diff
@@ -0,0 +1,108 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.AspNetCore.Components.RenderTree;
+using System;
+
+namespace Microsoft.AspNetCore.Components.Forms
+{
+    /// <summary>
+    /// An input component for editing date values.
+    /// Supported types are <see cref="DateTime"/> and <see cref="DateTimeOffset"/>.
+    /// </summary>
+    public class InputDate<T> : InputBase<T>
+    {
+        const string dateFormat = "yyyy-MM-dd"; // Compatible with HTML date inputs
+
+        [Parameter] string ParsingErrorMessage { get; set; } = "The {0} field must be a date.";
+
+        /// <inheritdoc />
+        protected override void BuildRenderTree(RenderTreeBuilder builder)
+        {
+            base.BuildRenderTree(builder);
+            builder.OpenElement(0, "input");
+            builder.AddAttribute(1, "type", "date");
+            builder.AddAttribute(2, "class", FieldClass);
+            builder.AddAttribute(3, "value", BindMethods.GetValue(CurrentValueAsString));
+            builder.AddAttribute(4, "onchange", BindMethods.SetValueHandler(__value => CurrentValueAsString = __value, CurrentValueAsString));
+            builder.CloseElement();
```

Comment:
Since these are implemented in C# it is totally possible for us to support splatting right? (not asking for you to add it to this PR)

Reply:
Yes, we can implement it in the built-in rendering, but that leads to a cliff edge for anyone who wants to subclass (say) `InputText` to set up custom rendering. They would lose the spatting, until we implement a built-in system for them to use. So until we do that I'd rather not have the feature in the base class.

Reply:
Fair enough